### ISSUE TITLE
Rule proposal: `prefer-array-index-of`

### ISSUE THREAD

**Issue by Fabrice-TIERCELIN** [sindresorhus/eslint-plugin-unicorn#919]
All built-ins have [`Array#indexOf()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/indexOf) in addition to [`Array#findIndex()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex). Prefer `Array#indexOf()` over use of `Array#findIndex()` passing a function that tests equality as parameter.

This rule is fixable.

## Fail

```js
[].findIndex(x => x === 'foo');
values.findIndex(x => 'foo' === x);
values.findIndex(x => {return x === 'foo';});
['foobar'].findIndex(x => {return 'foo' === x;});
values.findIndex(function (x) {return x === 'foo';});
['foobar'].findIndex(function (x) {return 'foo' === x;});
```

## Pass

```js
[].findIndex(i => i === list[i]);
[].findIndex(x => x === 'foo' && isValid);
[].findIndex(x => y === 'foo');
[].findIndex(x => y.x === 'foo');
[].findIndex(x => x == 'foo');
['foobar'].findIndex(function (x) {return 'foo' == x;});
[].findIndex(x => x !== 'foo');
values.findIndex(function (x) {return x !== 'foo';});
```

**Comment by fisker** [comment]
I think it's good.

**Comment by sindresorhus** [comment]
Accepted

**Comment by Fabrice-TIERCELIN** [comment]
I will try to use matchers.

**Comment by fisker** [comment]
We should name it `prefer-array-index-of`.

**Comment by Fabrice-TIERCELIN** [comment]
I think I will let you convert to esquery. It looks very powerful but not so easy from scratch. I don't know how to:

- Check that two nodes are the same like `bar` in `bar => bar === foo`,
- Combine several types of paths at the same time,
- Select after some functional checks,
- Retrieve several nodes from the selection (or I should proceed in another way).

**Comment by fisker** [comment]
I'll take a look.
